.github/workflows/.github-ci.yml

```diff
@@ -0,0 +1,41 @@
+name: Node.js CI
+
+# Run on every push or pull request to the 'main' branch (you can change this to your preference)
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [18.x]  # You can adjust the Node.js version according to your project
+
+    steps:
+      # Checkout the repository
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      # Set up Node.js
+      - name: Setup Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      # Install dependencies
+      - name: Install dependencies
+        run: npm install
+
+      # Step to run tests
+      - name: Run tests
+        run: npm test
+
+      # Step to run linter (ESLint and Prettier)
+      - name: Run linter
+        run: npm run lint
```

Comment:
The main advantage of pre-commit is that they "don't let you" push if your changes break the linting rules defined. Now we need to wait until the tests run in CI, which is less convenient. Could be combine those?

Reply:
We can cover this problem using the husky library, running automatically tests and eslint before push each commit.
Check package: [Husky docu](https://typicode.github.io/husky/)